automated_packaging/common_functions.pm

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/perl
+
+package common_functions;
+use Exporter;
+use JSON;
+
+# Export subroutines
+our @ISA= qw( Exporter );
+our @EXPORT = qw(get_and_verify_token update_configuration_file get_sorted_prs create_release_changelog has_backport_label);
+
+# untaint environment
+local $ENV{'PATH'} = '/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin';
+sub get_and_verify_token {
+    unless ( exists $ENV{GITHUB_TOKEN} ) {
+		die "You must have a GITHUB_TOKEN set";
+    }
+
+    my $github_token = $ENV{GITHUB_TOKEN};
+    if ( $ENV{GITHUB_TOKEN} =~ /^(\w+)$/x ) {
+        $github_token = $1;
+    }
+    else {
+    	die "Malformed GITHUB_TOKEN: $github_token";
+    }
+
+    my $cmd = "curl -sf -H 'Authorization: token $github_token' "
+      . 'https://api.github.com/';
+    my $result    = `$cmd > /dev/null 2>&1`;
+    my $exit_code = $? >> 8;
+
+    if ( $exit_code == 22 ) {
+    	die "Your token was rejected by GitHub.";
+    }
+
+    return $github_token;
+}
+
+# Update a file given old and new version strings
+sub update_configuration_file {
+    $new_version=@_[0];
+    $old_version_non_regex=@_[1];
+    $file_name=@_[2];
+
+    # Update configuration file
+    open(CONFIGURATION_FILE, "<$file_name") || die "$file_name not found";
+    my @lines = <CONFIGURATION_FILE>;
+    close(CONFIGURATION_FILE);
+
+    my @newlines;
+    foreach(@lines) {
+       $_ =~ s/$old_version_non_regex/$new_version/g;
+       push(@newlines,$_);
+    }
+
+    open(CONFIGURATION_FILE, ">$file_name") || die "$file_name not found";
+    print CONFIGURATION_FILE @newlines;
+    close(CONFIGURATION_FILE);
+}
+
+# Get sorted PRs according to merged-at by collecting PRs up to given data (using created info)
+sub get_sorted_prs {
+    my $last_date = @_[0];
+    my $repo_name = @_[1];
+    my %sorted_pr_hash = ();
+    my $github_token = get_and_verify_token();
+
+    my $page_number=1;
+    do {
+        my $prs_text = `curl -H "Accept: application/vnd.github.v3.full+json" -X GET --user "$github_token:x-oauth-basic" 'https://api.github.com/repos/citusdata/$repo_name/pulls?base=master&state=all&page=$page_number'`;
+        my @prs = @{decode_json($prs_text)};
+
+        foreach my $pr ( @prs ) {
+            my %pr_hash = %$pr;
+
+            if (defined($pr_hash{'merged_at'})) {
+
+                $merged_date = substr($pr_hash{'merged_at'}, 0 ,10);
+
+                if ($merged_date lt $last_date) {
+                    last;
+                }
+
+                $sorted_pr_hash{$pr_hash{'merged_at'}} = $pr_hash{'url'};
+            }
+        }
+
+        $page_number += 1;
+    } while($merged_date gt $last_date);
```

Comment:
Small possibility but if none of the PRs in first page is merged, `merged_date` will be undefined at this point.